docs/sources/operations/storage/wal.md

```diff
@@ -95,6 +95,18 @@ Statefulsets are significantly more cumbersome to work with/upgrade/etc. Much of
 
 In this case, try `kubectl -n <namespace> delete sts ingester --cascade=false`. This will leave the pods alive but delete the statefulset. Then you may recreate the (updated) statefulset and one-by-one start deleting the `ingester-0` through `ingester-n` pods _in that order_, allowing the statefulset to spin up new pods to replace them.
 
+#### Scaling Down Using `/flush_shutdown` Endpoint and Lifecycle Hook
+
+1. **StatefulSets for Ordered Scaling Down**: Loki's ingesters should be scaled down one by one, which is efficiently handled by Kubernetes StatefulSets. This ensures an ordered and reliable scaling process, as described in the [Deployment and Scaling Guarantees](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#deployment-and-scaling-guarantees) documentation.
+
+2. **Using PreStop Lifecycle Hook**: During the pod scaling down process, the PreStop [lifecycle hook](https://pwittrock.github.io/docs/concepts/containers/container-lifecycle-hooks/) triggers the `/flush_shutdown` endpoint on the ingester. This action flushes the chunks and removes the ingester from the ring, allowing it to register as unready and become eligible for deletion.
+
+3. **ScaleDown Policy**: The ScaleDown policy is configured to permit the removal of one pod every 30 minutes. This duration provides time for the ingester to flush its data before being deleted.
```

Comment:
This is a little misleading. The `terminationGracePeriodSeconds` is what provides time for the ingester to flush its data before being deleted.

https://github.com/grafana/loki/blob/425736614c43f3435e52bc1e5a2295dfcbb6418a/production/helm/loki/values.yaml#L679-L682

If you think flushing data may take 30 minutes, you might recommend increasing the `terminationGracePeriodSeconds`.

The scaledown policy is more important to avoid thrashing and for queries when configured to only query the ingesters. If all replicas of a stream are flushed from ingesters due to multiple pods scaledown within the `max_chunk_age`, the query will rely on the store to return the data. My understanding is that default Loki configuration handles this gracefully, so it would only be a concern with an odd config like `query_ingester_only: true`.